src/bytes.ts

```diff
@@ -1,19 +1,20 @@
-import { bytesToHex } from 'ethereum-cryptography/utils';
+import type { BytesLike, Hexable } from "@ethersproject/bytes";
 
-export type Bytes = Uint8Array;
+type Hex = BytesLike | Hexable | number | bigint;
```

Comment:
I think this should have a different name. Its `Hexlifyable`, or `Hexable`, but not `Hex`

Reply:
Note that, while its true that `toHex` supports more types than just `BytesLike`, it feels to me like this wyde `Hex` type is not used anywhere but here.

I would restrict `compare` to `BytesLike` because that is all that we actually need in this repo, and remove that type definition.